### ISSUE TITLE
CDK: Create a generic method for transforming records fields in accordance with streams schemas

### ISSUE THREAD

**Issue by Zirochkaa** [airbytehq/airbyte#5224]
## Tell us about the problem you're trying to solve
Right now we have few issues where fields types in records don't match types from streams schemas (#4292, #4841, #4827).

## Describe the solution you’d like
It would be nice to have a generic solution, a function or a class (like `Transformer` class in [singer](https://github.com/singer-io/singer-python/blob/66688d736fa699beeed63af91db2fdc1f8914bb1/singer/transform.py#L93)), which will go through each field in records (through nested fields too) and transform types to the ones in streams schemas.

## Describe the alternative you’ve considered or used
Right now if we want to change type of fields in records we are using `transform()` function (like in [source-github](https://github.com/airbytehq/airbyte/blob/766024591572801d3594fceab5cfe8cd549cbe33/airbyte-integrations/connectors/source-github/source_github/streams.py#L155)). In this case we need to manually specify fields and desired types. It works good for root keys, but for nested keys it's not so convenient to use such method.

## Additional context
First edition of this task will be implemented in #4827 issue.

## Are you willing to submit a PR?
Yes

**Comment by Zirochkaa** [comment]
Blocked on #4827.

**Comment by keu** [comment]
@Zirochkaa how is this issue blocked by #4827 ?

**Comment by Zirochkaa** [comment]
@keu we decided to do the first test version of this generic method in Facebook Marketing source in terms of #4827 issue, so this means that I need to add generic function to Facebook Marketing source first, merge it and only then I can move that generic function to CDk (maybe add some improvements).

**Comment by keu** [comment]
ok, lets call it playground
